### PR TITLE
5 refactor endpoint

### PR DESCRIPTION
**Pull request by odellmac4**
## Type of change
- [ ] New feature
- [ ] Bug Fix
- [ ] Test
- [x] Refactor/Style

## Description
Relevant User Story(s): 
-**Sad Path: endpoint 5**
*Brief summary of the change(s) included in this pull request*
-**Created validations for vendor model**
-**Created serializer method in error serializer to return 400 error when creation of object does not pass validation**
-**Captured error message from RecordInvalid and created method in ApplicationController  to return custom json response**

## Implements/Fixes:

description closes #

## Check the correct boxes
- [x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Testing Changes
- [ ] No Tests have been changed
- [x] Some Tests have been changed
- [ ] All of the Tests have been changed

*Please describe what in the world happened if tests are broken*

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

### CODE REVIEW

File: spec/requests/api/v0/vendors_request_spec.rb [hunk 1]
```diff
@@ -45,7 +45,7 @@
 
   describe "delete a Vendor" do
     it "can destroy a Vendor from the database" do
-      vendor = create(:vendor)
+      vendor = create(:vendor, credit_accepted: false)
```
Comment (yainporter) [inline spec/requests/api/v0/vendors_request_spec.rb:49]: Is there a reason why we are explicitly stating the :credit_accepted value?

File: spec/requests/api/v0/vendors_request_spec.rb [hunk 2]
```diff
@@ -59,7 +59,7 @@
 
       ### Alternate test to check for destroyed Vendor
 
-      vendor2 = create(:vendor)
+      vendor2 = create(:vendor, credit_accepted: true)
```
Comment (yainporter) [inline spec/requests/api/v0/vendors_request_spec.rb:63]: Is there a reason why we are explicitly stating the :credit_accepted value?

**Review by yainporter — APPROVED** [review]
Nice work on figuring out how to raise that validation error!

File: spec/requests/api/v0/vendors_request_spec.rb (hunk 2 above)
Comment (odellmac4) [reply spec/requests/api/v0/vendors_request_spec.rb:63]: sorry I did when testing my validations but we do not need this anymore or the commented out validation in the vendor model. I am going to push those changes up now

**Pull request merged**